general/optparser.cpp

```diff
@@ -248,6 +248,25 @@ void OptionsParser::Parse()
    error_type = 0;
 }
 
+void OptionsParser::ParseCheck(std::ostream &out)
+{
+   Parse();
+   if (!Good())
+   {
+      PrintUsage(out);
```

Comment:
This will print from all ranks in parallel.

Reply:
Good point. I think I fixed it now.